include/ap/argument_parser.hpp

```diff
@@ -11,6 +11,7 @@
 #include <string_view>
 #include <type_traits>
 #include <vector>
+#include <set>
```

Comment:
To ensure that the set can be created properly add a `less_comparable` concept in the `ap::utility` namespace, which requires a type to overload the less `<` operator.

As a second option you can create a template `choices` function which would taka a `Compare` as a template parameter. Then you could initialize the choices set as `std::set<value_type, Compare>`. Here the for the `Compare` predicate you would have to add a new concept - `binary_predicate<T>` to enforce the correctness of the template parameter.

I also think that the choices functions do not have to be defined in the inerface as they are not used in the parser, but rather by a user, for whom it would be easier to pass a `std::vector<value_type>` rather than a `std::vector<std::any>`